server/src/main/java/org/elasticsearch/search/query/QueryPhaseCollector.java

```diff
@@ -300,9 +336,9 @@ public void collect(int doc) throws IOException {
                     }
                 }
             }
-            // min_score is applied to aggs as well as top hits
-            if (isDocWithinMinScore(scorer)) {
-                if (aggsLeafCollector != null) {
+            if (aggsLeafCollector != null) {
+                // min_score is applied to aggs as well as top hits
+                if (isDocWithinMinScore(scorer)) {
```

Comment:
Did you mean to swap the order of these if statements?

Reply:
yes, we don't need to check the min score if we are not going to collector from the aggs leaf collector.